src/main/java/gtPlusPlus/core/handler/events/PlayerSleepEventHandler.java

```diff
@@ -0,0 +1,106 @@
+package gtPlusPlus.core.handler.events;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import cpw.mods.fml.common.eventhandler.SubscribeEvent;
+import gtPlusPlus.api.objects.Logger;
+import gtPlusPlus.core.potion.GtPotionEffect;
+import gtPlusPlus.core.util.Utils;
+import gtPlusPlus.core.util.math.MathUtils;
+import gtPlusPlus.core.util.minecraft.PlayerUtils;
+import gtPlusPlus.core.util.reflect.ReflectionUtils;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.potion.Potion;
+import net.minecraft.potion.PotionEffect;
+import net.minecraft.util.ChatComponentTranslation;
+import net.minecraftforge.event.entity.player.PlayerSleepInBedEvent;
+import net.minecraftforge.event.entity.player.PlayerWakeUpEvent;
+
+public class PlayerSleepEventHandler {
+
+	private static Field sEffectDuration = ReflectionUtils.getField(PotionEffect.class, "duration");
+	private static ArrayList<Potion> sPositiveEffects = new ArrayList<Potion>();
+	private static ArrayList<Potion> sNegativeEffects = new ArrayList<Potion>();
```

Comment:
Any reason not to just use a `Map<int, Potion>` and look it up instead of iterate over it for every active effect?

Suppose it's not that much iteration just... feels dirty.